service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -26,7 +34,60 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         Object migrationId = caseDetails.getData().get(MIGRATION_ID_KEY);
 
+        if ("FPLA-2982".equals(migrationId)) {
+            run2982(caseDetails);
+        }
+
         caseDetails.getData().remove(MIGRATION_ID_KEY);
         return respond(caseDetails);
     }
+
+    private void run2982(CaseDetails caseDetails) {
+        final List<Long> validCases = List.of(
+            1598429153622508L,
+            1615191831533551L,
+            1594384486007055L,
+            1601977974423857L,
+            1615571327261140L,
+            1615476016828466L,
+            1616507805759840L,
+            1610015759403189L,
+            1615994076934396L,
+            1611613172339094L,
+            1612440806991994L,
+            1607004182103389L,
+            1617045146450299L,
+            1612433400114865L,
+            1615890702114702L,
+            1610018233059619L
+        );
+
+        CaseData caseData = getCaseData(caseDetails);
+
+        if (!validCases.contains(caseData.getId())) {
+            throw new IllegalArgumentException("Invalid case Id");
+        }
+
+        List<Element<AdditionalApplicationsBundle>> additionalApplicationsBundle =
+            caseData.getAdditionalApplicationsBundle();
+
+        if (additionalApplicationsBundle.stream()
+            .noneMatch(bundle -> bundle.getValue().getC2DocumentBundle().getId() == null)) {
```

Comment:
we have to check for otherApplicationBundle as well
`bundle.getValue().getOtherApplciationBundle().getId()`

Reply:
This has now been incorporated as discussed